src/main/java/org/kohsuke/github/GHCheckRunBuilder.java

```diff
@@ -0,0 +1,370 @@
+    public static final class Annotation {
+
+        private final transient Output output;
+        private final String path;
+        private final int start_line;
+        private final int end_line;
+        private final String annotation_level;
+        private final String message;
+        private Integer start_column;
+        private Integer end_column;
+        private String title;
+        private String raw_details;
+
+        Annotation(Output output,
+                String path,
+                int start_line,
+                int end_line,
+                GHCheckRun.AnnotationLevel annotation_level,
+                String message) {
+            this.output = output;
+            this.path = path;
+            this.start_line = start_line;
+            this.end_line = end_line;
+            this.annotation_level = annotation_level.toString().toLowerCase(Locale.ROOT);
+            this.message = message;
+        }
+
+        public @NonNull Annotation withStartColumn(@CheckForNull Integer startColumn) {
+            start_column = startColumn;
+            return this;
+        }
+
+        public @NonNull Annotation withEndColumn(@CheckForNull Integer endColumn) {
+            end_column = endColumn;
+            return this;
+        }
+
+        public @NonNull Annotation withTitle(@CheckForNull String title) {
+            this.title = title;
+            return this;
+        }
+
+        public @NonNull Annotation withRawDetails(@CheckForNull String rawDetails) {
+            raw_details = rawDetails;
+            return this;
+        }
+
+        public @NonNull Output done() {
+            if (output.annotations == null) {
+                output.annotations = new LinkedList<>();
+            }
+            output.annotations.add(this);
+            return output;
+        }
+
+    }
+
+    /**
+     * @see <a href="https://developer.github.com/v3/checks/runs/#images-object">documentation</a>
+     */
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public static final class Image {
+
+        private final transient Output output;
+        private final String alt;
+        private final String image_url;
+        private String caption;
+
+        Image(Output output, String alt, String image_url) {
+            this.output = output;
+            this.alt = alt;
+            this.image_url = image_url;
+        }
+
+        public @NonNull Image withCaption(@CheckForNull String caption) {
+            this.caption = caption;
+            return this;
+        }
+
+        public @NonNull Output done() {
+            if (output.images == null) {
+                output.images = new LinkedList<>();
+            }
+            output.images.add(this);
+            return output;
+        }
+
+    }
+
+    /**
+     * @see <a href="https://developer.github.com/v3/checks/runs/#actions-object">documentation</a>
+     */
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public static final class Action {
```

Comment:
Does this need to be `public`?  It is never needed by users.

Reply:
I was leaving it `public` in case the Jackson databinding required it to be, but can check if tests still pass if it is made less accessible.